pkg/compiler/validators/typing.go

```diff
@@ -130,6 +134,46 @@ func (t schemaTypeChecker) CastsFrom(upstreamType *flyte.LiteralType) bool {
 	return true
 }
 
+// Structured dataset are more complex types in the Flyte ecosystem. A structured dataset is considered castable in the following
+// cases.
+//
+//    1. The downstream structured dataset has no column types specified.  In such a case, it accepts all structured dataset input since it is
+//       generic.
+//
+//    2. The downstream structured dataset has a subset of the upstream columns and they match perfectly.
+//
+func (t structuredDatasetChecker) CastsFrom(upstreamType *flyte.LiteralType) bool {
+	// structured dataset are nullable
+	if isVoid(upstreamType) {
```

Comment:
is this common?

Reply:
we did the same thing in schema, so I added this